src/models/Submission.js

```diff
@@ -1,93 +1,87 @@
-			allowNull    : true
-		},
+    userId: {
+      type: DataTypes.INTEGER,
+      allowNull: true
+    },
+    formName: {
+      type: DataTypes.STRING,
+      allowNull: true
+    },
+    status: {
+      type: DataTypes.ENUM('approved', 'pending', 'unapproved'),
+      defaultValue: 'pending',
+      allowNull: false
+    },
 
-		status: {
-			type         : DataTypes.ENUM('approved','pending','unapproved'),
-			defaultValue : 'pending',
-			allowNull    : false
-		},
+    submittedData: {
+      type: DataTypes.TEXT,
+      allowNull: false,
+      defaultValue: {},
+      get: function () {
+        // for some reaason this is not always done automatically
+        let value = this.getDataValue('submittedData');
+        try {
+          if (typeof value == 'string') {
+            value = JSON.parse(value);
+          }
+        } catch (err) {
+        }
+        return value;
+      },
+      set: function (value) {
+        try {
+          if (typeof value == 'string') {
+            value = JSON.parse(value);
+          }
+        } catch (err) {
+        }
 
-		submittedData: {
-			type         : DataTypes.TEXT,
-			allowNull    : false,
-			defaultValue : {},
-			get          : function() {
-				// for some reaason this is not always done automatically
-				let value = this.getDataValue('submittedData');
-				try {
-					if (typeof value == 'string') {
-						value = JSON.parse(value);
-					}
-				} catch(err) {}
-				return value;
-			},
-			set: function(value) {
-				try {
-					if (typeof value == 'string') {
-						value = JSON.parse(value);
-					}
-				} catch(err) {}
-				let newValue = {};
+        this.setDataValue('submittedData', JSON.stringify(value));
+      }
+    },
 
-				value = value;
-
-				/*
-				const configExtraData = [];
-				if (configExtraData) {
-					Object.keys(configExtraData).forEach((key) => {
-						if (configExtraData[key].allowNull === false && (typeof value[key] === 'undefined' || value[key] === '')) { // TODO: dit wordt niet gechecked als je het veld helemaal niet meestuurt
-							// zie validExtraData hieronder
-							// throw db.sequelize.ValidationError(`${key} is niet ingevuld`);
-						}
-						if (value[key] && configExtraData[key].values.indexOf(value[key]) != -1) { // TODO: alles is nu enum, maar dit is natuurlijk veel te simpel
-							newValue[key] = value[key];
-						}
-					});
-				}
-				*/
-
-				this.setDataValue('submittedData', JSON.stringify(value));
-			}
-		},
+  }, { hooks: {
+    afterCreate: (instance, options) => {
+      const ruleSetData = {
+        resource: 'submission',
+        eventType: 'CREATE',
+        instance
+      }
+      eventService.publish(db, parseInt(instance.siteId), ruleSetData);
+    }
+  }
+  });
 
-	});
+  Submission.scopes = function scopes() {
+    return {
+      defaultScope: {},
+      withUser: {
+        include: [{
+          model: db.User,
+          attributes: ['role', 'nickName', 'firstName', 'lastName', 'email']
```

Comment:
Voor sommige sites willen we de achternaam niet tonen, wellicht eens nadenken of we dit configurabel kunnen maken bij de scopes, ipv het er later uitslopen.

Reply:
Of kan het wellicht achter zo'n viewableBy check?